src/displayapp/screens/Dice.cpp

```diff
@@ -0,0 +1,231 @@
+  auto dice = static_cast<Dice*>(obj->user_data);
+  dice->numDiceDecBtnEventHandler(event);
+}
+
+static void dice_size_inc_event_handler(lv_obj_t* obj, lv_event_t event) {
+  auto dice = static_cast<Dice*>(obj->user_data);
+  dice->diceSizeIncBtnEventHandler(event);
+}
+
+static void dice_size_dec_event_handler(lv_obj_t* obj, lv_event_t event) {
+  auto dice = static_cast<Dice*>(obj->user_data);
+  dice->diceSizeDecBtnEventHandler(event);
+}
+
+static void roll_event_handler(lv_obj_t* obj, lv_event_t event) {
+  auto dice = static_cast<Dice*>(obj->user_data);
+  dice->rollBtnEventHandler(event);
+}
+
+void Dice::createScreen() {
+  resultCont = lv_cont_create(lv_scr_act(), nullptr);
+  lv_obj_set_size(resultCont, 240, 80);
+  lv_obj_set_pos(resultCont, 0, 0);
+  lv_cont_set_layout(resultCont, LV_LAYOUT_COLUMN_LEFT);
+  lv_obj_set_style_local_pad_all(resultCont, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, 10);
+
+  resultTotalLabel = lv_label_create(resultCont, nullptr);
+  lv_obj_set_size(resultTotalLabel, 220, 30);
+  lv_obj_set_style_local_text_color(resultTotalLabel, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+  lv_label_set_text(resultTotalLabel, "");
+
+  resultIndRollsLabel = lv_label_create(resultCont, nullptr);
+  lv_label_set_long_mode(resultIndRollsLabel, LV_LABEL_LONG_SROLL_CIRC);
+  lv_obj_set_size(resultIndRollsLabel, 220, 30);
+  lv_obj_set_style_local_text_color(resultIndRollsLabel, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+  lv_label_set_text(resultIndRollsLabel, "");
+
+  configCont = lv_cont_create(lv_scr_act(), nullptr);
+  lv_obj_set_size(configCont, 240, 160);
+  lv_obj_set_pos(configCont, 0, 80);
+  lv_obj_set_style_local_bg_color(configCont, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+
+  rollBtn = lv_btn_create(configCont, nullptr);
+  rollBtn->user_data = this;
+  lv_obj_set_event_cb(rollBtn, roll_event_handler);
+  lv_obj_set_size(rollBtn, 70, 150);
+  lv_obj_set_pos(rollBtn, 165, 5);
+  lv_obj_set_ext_click_area(rollBtn, 5, 5, 5, 5);
+  rollBtnLabel = lv_label_create(rollBtn, nullptr);
+  lv_label_set_text(rollBtnLabel, "Roll");
+
+  numDiceIncBtn = lv_btn_create(configCont, nullptr);
+  numDiceIncBtn->user_data = this;
+  lv_obj_set_event_cb(numDiceIncBtn, num_dice_inc_event_handler);
+  lv_obj_set_size(numDiceIncBtn, 70, 50);
+  lv_obj_set_pos(numDiceIncBtn, 5, 5);
+  lv_obj_set_ext_click_area(numDiceIncBtn, 5, 5, 5, 10);
+  numDiceIncBtnLabel = lv_label_create(numDiceIncBtn, nullptr);
+  lv_label_set_text(numDiceIncBtnLabel, "+");
+
+  numDiceLabelCont = lv_cont_create(configCont, nullptr);
+  lv_obj_set_size(numDiceLabelCont, 80, 40);
+  lv_obj_set_pos(numDiceLabelCont, 0, 60);
+  lv_obj_set_style_local_bg_color(numDiceLabelCont, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+  lv_cont_set_layout(numDiceLabelCont, LV_LAYOUT_CENTER);
+  numDiceLabel = lv_label_create(numDiceLabelCont, nullptr);
+  lv_label_set_text_fmt(numDiceLabel, "%d", 1);
```

Comment:
If you're formatting with a constant value you might as well just `lv_label_set_text_static(numDiceLabel, "1");` instead and avoid the cost of formatting. Same applies to setting `diceSizeLabel` below.

Reply:
@jonvmey Fixed this in the latest changes.